app/celery/contact_information_tasks.py

```diff
@@ -1,18 +1,134 @@
+from typing import Optional
+from sqlalchemy.orm.exc import NoResultFound
 from flask import current_app
 from app import notify_celery, va_profile_client
 from app.celery.common import can_retry, handle_max_retries_exceeded
 from app.celery.exceptions import AutoRetryException
 from app.celery.service_callback_tasks import check_and_queue_callback_task
+from app.feature_flags import FeatureFlag, is_feature_enabled
 from app.va.identifier import IdentifierType
-from app.va.va_profile import VAProfileRetryableException, VAProfileNonRetryableException, NoContactInfoException
+from app.va.va_profile import (
+    VAProfileRetryableException,
+    VAProfileNonRetryableException,
+    NoContactInfoException,
+    VAProfileResult,
+)
+from app.dao.communication_item_dao import get_communication_item
 from app.dao.notifications_dao import get_notification_by_id, dao_update_notification, update_notification_status_by_id
-from app.models import NOTIFICATION_PERMANENT_FAILURE, EMAIL_TYPE, SMS_TYPE
+from app.models import (
+    NOTIFICATION_PERMANENT_FAILURE,
+    NOTIFICATION_PREFERENCES_DECLINED,
+    EMAIL_TYPE,
+    SMS_TYPE,
+    RecipientIdentifier,
+)
 from app.exceptions import NotificationTechnicalFailureException, NotificationPermanentFailureException
-from app.va.va_profile.exceptions import VAProfileIDNotFoundException
+from app.va.va_profile.exceptions import (
+    VAProfileIDNotFoundException,
+    CommunicationItemNotFoundException,
+)
 from notifications_utils.statsd_decorators import statsd
 from requests import Timeout
 
 
+def get_recipient(notification_type: str, notification_id: str, recipient_identifier: RecipientIdentifier) -> str:
```

Comment:
this is inconsistent throughout all files in this PR, but consider adding a trailing comma so that the args are broken out one per line. right now, some are broken out, some are not. I find it more readable when broken out.

Reply:
Trailing comma added